cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -325,25 +326,46 @@ class AesDecryptor::AesDecryptorImpl {
     }
   }
 
-  int ciphertext_size_delta() { return ciphertext_size_delta_; }
+  [[nodiscard]] int PlaintextLength(int ciphertext_len) const {
+    if (ciphertext_len < ciphertext_size_delta_) {
+      std::stringstream ss;
+      ss << "Ciphertext length " << ciphertext_len << " is invalid, expected at least "
+         << ciphertext_size_delta_;
+      throw ParquetException(ss.str());
+    }
+    return ciphertext_len - ciphertext_size_delta_;
+  }
+
+  [[nodiscard]] int CiphertextLength(int plaintext_len) const {
+    if (plaintext_len < 0) {
+      std::stringstream ss;
+      ss << "Negative plaintext length " << plaintext_len;
+      throw ParquetException(ss.str());
+    }
+    return plaintext_len + ciphertext_size_delta_;
```

Comment:
We should ideally check for signed addition overflow here.

Reply:
Good point. Is it ok to add this to the changes in #43195, or should I make a separate PR for these follow ups?

Reply:
IMHO you can add them to the followup PR.

Reply:
I've added this check in https://github.com/apache/arrow/pull/43195/commits/fb63bcc7456a401b81f5fbacd6b728550c0d788e